pkg/daemon/controller/controller.go

```diff
@@ -167,30 +200,31 @@ func (sc *ServiceController) Reconcile(_ context.Context, req ctrl.Request) (res
 		return ctrl.Result{}, err
 	}
 
-	_, present := node.Annotations[desiredVersionAnnotation]
+	desiredVersion, present := node.Annotations[desiredVersionAnnotation]
 	if !present {
 		// node missing desired version annotation, don't requeue
 		return ctrl.Result{}, nil
 	}
 
-	// TODO: When permission issues are resolved, retrieve ConfigMap itself, instead of using this annotation
-	encodedCMData, present := node.Annotations[servicescm.CMDataAnnotation]
-	if !present {
-		// node missing CMData annotation, don't requeue
-		return ctrl.Result{}, nil
+	// Fetch the CM of the desired version
+	var cm core.ConfigMap
+	err = sc.client.Get(sc.ctx, client.ObjectKey{Namespace: sc.watchNamespace, Name: servicescm.NamePrefix + desiredVersion}, &cm)
+	if err != nil {
```

Comment:
Looks like this can be condensed into the one line conditional error check format.